### PR TITLE
BI-489 - ConcurrentModificationException during concurrent artifact d…

### PR DESCRIPTION
**Pull request by yahavi**
ConcurrentModificationException during concurrent artifact download.

### CODE REVIEW

File: build-info-extractor/src/main/java/org/jfrog/build/extractor/clientConfiguration/util/DependenciesDownloaderHelper.java [hunk 1]
```diff
@@ -59,7 +60,7 @@ public DependenciesDownloaderHelper(DependenciesDownloader downloader, Log log)
         this.log = log;
     }
 
-    public DependenciesDownloaderHelper(ArtifactoryDependenciesClient client, String workingDirectory, Log log) {
+    public DependenciesDownloaderHelper(ArtifactoryDependenciesClientBuilder client, String workingDirectory, Log log) {
```
Comment (barbelity) [inline build-info-extractor/src/main/java/org/jfrog/build/extractor/clientConfiguration/util/DependenciesDownloaderHelper.java]: Renaming the received object to "clientBuilder" may be a bit more meaningful.

File: build-info-extractor/src/main/java/org/jfrog/build/extractor/clientConfiguration/util/DependenciesDownloaderImpl.java [hunk 2]
```diff
@@ -26,21 +27,21 @@
...
 public class DependenciesDownloaderImpl implements DependenciesDownloader {
 
-    private ArtifactoryDependenciesClient client;
+    private ArtifactoryDependenciesClientBuilder client;
     private File workingDirectory;
     private Log log;
     private boolean flatDownload = false;
 
-    public DependenciesDownloaderImpl(ArtifactoryDependenciesClient client, String workingDirectory, Log log) {
+    public DependenciesDownloaderImpl(ArtifactoryDependenciesClientBuilder client, String workingDirectory, Log log) {
```
Comment (barbelity) [inline build-info-extractor/src/main/java/org/jfrog/build/extractor/clientConfiguration/util/DependenciesDownloaderImpl.java]:
```suggestion
    public DependenciesDownloaderImpl(ArtifactoryDependenciesClientBuilder clientBuilder, String workingDirectory, Log log) {
```

File: build-info-extractor/src/main/java/org/jfrog/build/extractor/clientConfiguration/util/DependenciesDownloaderImpl.java [hunk 3]
```diff
@@ -26,21 +27,21 @@
  */
 public class DependenciesDownloaderImpl implements DependenciesDownloader {
 
-    private ArtifactoryDependenciesClient client;
+    private ArtifactoryDependenciesClientBuilder client;
```
Comment (barbelity) [inline build-info-extractor/src/main/java/org/jfrog/build/extractor/clientConfiguration/util/DependenciesDownloaderImpl.java]:
```suggestion
    private ArtifactoryDependenciesClientBuilder clientBuilder;
```

File: build-info-extractor/src/main/java/org/jfrog/build/extractor/clientConfiguration/util/spec/SpecsHelper.java [hunk 4]
```diff
@@ -137,7 +138,7 @@ public SpecsHelper(Log log) {
      * @return A list of the downloaded dependencies.
      * @throws IOException in case of IOException
      */
-    public List<Dependency> downloadArtifactsBySpec(String spec, ArtifactoryDependenciesClient client, String targetDirectory) throws IOException {
+    public List<Dependency> downloadArtifactsBySpec(String spec, ArtifactoryDependenciesClientBuilder client, String targetDirectory) throws IOException {
```
Comment (barbelity) [inline build-info-extractor/src/main/java/org/jfrog/build/extractor/clientConfiguration/util/spec/SpecsHelper.java]:
```suggestion
    public List<Dependency> downloadArtifactsBySpec(String spec, ArtifactoryDependenciesClientBuilder clientBuilder, String targetDirectory) throws IOException {
```

File: build-info-extractor/src/main/java/org/jfrog/build/extractor/clientConfiguration/util/spec/SpecsHelper.java [hunk 5]
```diff
@@ -137,7 +138,7 @@ public SpecsHelper(Log log) {
...
         DependenciesDownloaderHelper helper = new DependenciesDownloaderHelper(client, targetDirectory, log);
```
Comment (barbelity) [inline build-info-extractor/src/main/java/org/jfrog/build/extractor/clientConfiguration/util/spec/SpecsHelper.java]:
```suggestion
        DependenciesDownloaderHelper helper = new DependenciesDownloaderHelper(clientBuilder, targetDirectory, log);
```

**Comment by yahavi** [comment]
https://github.com/jfrog/jenkins-artifactory-plugin/pull/163

**Pull request merged**